Documentation/FrameworkOverview.md

```diff
@@ -8,196 +8,219 @@ learning about new modules and finding more specific documentation.
-(for example, sending a network request).
+A **Buffer**, created by `SignalProducer.buffer()`, is a (optionally bounded)
+queue for [Events](#events) and replays those events when new 
+[Signals](#signals) are created from the producer.
 
-A connection is created through the `-publish` or `-multicast:` methods on
-[RACSignal][RACSignal+Operations], and ensures that only one underlying
-subscription is created, no matter how many times the connection is subscribed
-to. Once connected, the connection's signal is said to be _hot_, and the
-underlying subscription will remain active until _all_ subscriptions to the
-connection are [disposed](#disposables).
+Similar to a [Pipe](#pipes), the method returns an [Observer](#observers). 
+Events sent to this observer will be added to the queue. If the buffer is already
+at capacity, the earliest (oldest) event will be dropped to make room for the 
+new event. 
 
-## Sequences
+This can be usefull to buffer events for future observers, like when a network 
+request finishes before anything is ready to handle the result.
 
-A **sequence**, represented by the [RACSequence][] class, is a _pull-driven_
-[stream](#streams).
+## Observers
 
-Sequences are a kind of collection, similar in purpose to `NSArray`. Unlike
-an array, the values in a sequence are evaluated _lazily_ (i.e., only when they
-are needed) by default, potentially improving performance if only part of
-a sequence is used. Just like Cocoa collections, sequences cannot contain `nil`.
+An **Observer** is anything that is waiting or capable of waiting for events
+from a [signal](#signals). Within RAC, an observer is represented as an instance
+of the [SinkOf][] struct with an input type of Event.
 
-Sequences are similar to [Clojure's sequences][seq] ([lazy-seq][] in particular), or
-the [List][] type in [Haskell][].
+A signal can be observed by calling its `observe` method, providing either a
+sink or callbacks for the different types of events as a parameter.
+
+## Action
+
+An **Action**, represented by the [Action][] class, will do some work when
+executed with an _Input_. During or after execution, zero or more _Output_
+values are generated. Alternatively, an _Error_ may be generated.
+
+Actions are suited to perform side-effecting work as the user interacts with
+the app.
+
+Usually the trigger for an Action is UI-driven, like when a button is
+clicked. Actions can also be automatically disabled based on a signal, and this
+disabled state can be represented in a UI by disabling any controls associated
+with the action.
 
-RAC adds a `-rac_sequence` method to most of Cocoa's collection classes,
-allowing them to be used as [RACSequences][RACSequence] instead.
+For interaction with UIKit or AppKit GUI controls such as `NSControl` or 
+`UIControl`, RAC provides [CocoaAction][] to wrap Action with KVO or 
+Cocoa Bindings.
+
+## Property
+
+A **Property**, represented by the [PropertyType][Property] protocol, allows
+observation of its changes.
```

Comment:
An attempt to combine the paragraphs into one: 

> A **Property**, represented by the [PropertyType][Property] protocol, 
wraps a value, allowing observation of value changes via a 
[SignalProducer](#signal-producers) that can be obtained from the `producer` getter.
This signal producer will send the property`s current value, followed by all 
changes over time. Alternatively, the current value can be accessed via the
`value` getter.

I also like your first suggestion with the paragraphs as they were before, which one do you think is better?